tests/e2e/specs/NestedValidationRules.spec.js

```diff
@@ -0,0 +1,110 @@
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-parent_loop_input_2', 0);
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-parent_loop_input_2', 1);
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-parent_loop_input_2', 2);
+    shouldHaveValidationErrors();
+
+    fillInputText('screen-field-nested_1');
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-nested_2');
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-nested_3');
+
+    submitForm();
+  });
+
+  it('Verify validation rules with hidden fields before submit', () => {
+    cy.loadFromJson('nested_validations_with_hidden_rules.json', 0);
+    cy.get('[data-cy=mode-preview]').click();
+
+    cy.get('[data-cy=preview-content] [name="submit"]')
+      .click();
+
+    fillInputText('screen-field-parent_input_1');
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-parent_loop_input_2', 0);
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-parent_loop_input_2', 1);
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-parent_loop_input_2', 2);
+    shouldHaveValidationErrors();
+
+    fillInputText('screen-field-nested_1');
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-nested_2');
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-nested_3');
+
+    submitForm();
+  });
+
+  it('Verify validation rules with hidden fields in nested screen before submit', () => {
+    cy.loadFromJson('nested_validations_with_hidden_rules.json', 0);
+    cy.get('[data-cy=mode-preview]').click();
+
+    cy.get('[data-cy=preview-content] [name="submit"]')
+      .click();
+
+    fillInputText('screen-field-parent_input_1');
+    shouldHaveValidationErrors();
+
+    fillInputText('screen-field-nested_1');
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-nested_2');
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-nested_3');
+    shouldHaveValidationErrors();
+
+    fillInputText('screen-field-parent_loop_input_2', 0);
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-parent_loop_input_2', 1);
+    shouldHaveValidationErrors();
+    fillInputText('screen-field-parent_loop_input_2', 2);
+
+    submitForm();
+  });
+});
+
+function fillInputText(dataCy, index = null)
+{
+  if (index === null) {
+    cy.get(`[data-cy=preview-content] [data-cy="${dataCy}"]`)
+      .clear()
+      .type('test');
+  } else {
+    cy.get(`[data-cy=preview-content] [data-cy="${dataCy}"]`).eq(index)
+      .clear()
+      .type('test');
+  }
+}
+
+function shouldHaveValidationErrors()
+{
+  cy.get('[data-cy=preview-content] [data-cy="screen-field-submit"]')
+    .should('contain.html', 'alert alert-danger');
+}
+
+function submitForm()
+{
+  cy.get('[data-cy=preview-content] [data-cy="screen-field-submit"]')
+    .should('not.contain.html', 'alert alert-danger');
```

Comment:
```suggestion
    .should('have.length', 0);
```

Reply:
Since the 4.1 version does not show the invalid fields, chance this test to check if the form has invalid fields